packages/effect/src/optic/Optic.ts

```diff
@@ -368,8 +368,11 @@ function make(ast: AST.AST): any {
   }
 }
 
-function shallowCopy(s: object): any {
-  return Array.isArray(s) ? s.slice() : { ...s }
+/** @internal */
+export function cloneShallow<T>(x: T): T {
+  if (Array.isArray(x)) return x.slice() as T
+  if (typeof x === "object" && x !== null) return { ...x } as T
```

Comment:
How will this play out in the case of `.prototype`? I think it's reasonable to preserve it here

Reply:
@nikelborm it's intended to be used with POJOs: https://github.com/Effect-TS/effect-smol/blob/f23760590b838e309a46338d9875bedfdd064c55/packages/effect/SCHEMA.md#problem

Reply:
> Or maybe some runtime check that would reject objects that have a prototype other than object or null?

sure